core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -230,6 +231,21 @@ public String toString()
         }
     }
 
+    void setRowIndex(int rowIndex)
+    {
+        this.rowIndex = rowIndex;
```

Comment:
```
if (this.rowIndex != -1) {
  throw new IllegalStateException("...");
}
```

Reply:
No, we cannot do it because of:

```
    @Override
    public SingleRowBlockWriter beginBlockEntry()
    {
        if (currentEntryOpened) {
            throw new IllegalStateException("Expected current entry to be closed but was opened");
        }
        currentEntryOpened = true;
        singleRowBlockWriter.setRowIndex(fieldBlockBuilders[0].getPositionCount());
        return singleRowBlockWriter;
    }

```

Reply:
I don't see relationship?
Why `beginBlockEntry` prevents `if (this.rowIndex != -1) {` check?

Reply:
Ah, I see. You have put **this**.rowIndex. Sorry for misunderstanding.